statuslight_control/lib/statuslight_control/statuslight.ex

```diff
@@ -1,26 +1,38 @@
 defmodule StatuslightControl.Statuslight do
   use GenServer
+  @renderer Application.get_env(:statuslight_control, :renderer)
+  @pixel_width 12
+  @pixel_height 1
 
-  def start_link(renderer) do
-    GenServer.start_link(__MODULE__, renderer, name: StatusLight)
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, %{color: "green"}, name: __MODULE__)
   end
 
   def stop do
-    GenServer.stop(StatusLight)
+    GenServer.stop(__MODULE__)
   end
 
-  def init(renderer) do
-    {:ok, %{renderer: renderer} }
+  @impl true
```

Comment:
I wasn't familiar with this - from my reading of the docs it is a way to assert that this (in very OO terms) is overriding a function from GenServer in order to get stronger compile-time guarantees.  Is that accurate?

Reply:
I think it's not so much compile time guarantees as much a discovery/readability thing. I believe it's best practice becuase they use this all throughout the docs